modules/tabix/bgzip/main.nf

```diff
@@ -23,8 +24,9 @@ process TABIX_BGZIP {
     in_bgzip = input.toString().endsWith(".gz")
     command1 = in_bgzip ? '-d' : '-c'
     command2 = in_bgzip ? ''   : " > ${prefix}.${input.getExtension()}.gz"
+    gzi_args = args.matches("(^| )-i\\b") ? "-I ${prefix}.${input.getExtension()}.gz.gzi" : ''
     """
-    bgzip $command1 $args -@${task.cpus} $input $command2
+    bgzip $command1 $args $gzi_args -@${task.cpus} $input $command2
```

Comment:
I'd rather we use $args2 instead of $gzi_args as we do in other modules

Reply:
why do we need the second one in the first place? Can you not check the $args string for what you need?

Reply:
Good point @FriederikeHanssen : before adding `-I (...)` I should actually double-check it's not already there. And I should check both the short and long forms of the options.

Therefore, the feature becomes two-fold:

1. Capture the `.gzi` file if indexing has been requested (`-i` or `--index`)
2. Name the `.gzi` file according to the standard prefix unless a specific name has been requested (`-I` or `--index-name`)

I'm not keen on using `$args2` because the convention seems to be 1) it refers to the _second_ command of the script, whereas here it would be used in the _first_ command, and 2) it's always initialised from user input (`task.ext.args2`), whereas here we want to complement the existing arguments from `task.ext.args`.
With the latest version, I modify `$args` directly.